lib/ddtrace.rb

```diff
@@ -1,5 +1,11 @@
 # typed: strict
 
+# Load logging before tracing as trace <> log correlation,
+# implemented in the tracing component, depends on logging
+# data structures.
+# Logging is also does not depend on other products.
+require 'datadog/logging'
```

Comment:
I'm... not entirely convinced by this:

1) Creating a whole new top-level module just to contain a single ext with a single string saying "ruby" seems a lot of complexity for such a simple thing
2) If tracing depends on this, shouldn't it be required in `datadog/tracing`? This seems like it'll break for customers that use `require 'datadog/tracing'` instead of `require 'ddtrace'`.


I would suggest instead having this inside `lib/datadog/core/logging/ext.rb` or something similar. We can always move it out once a bigger "logging" thing starts taking shape, but I think it's a bit too early to do so.

Reply:
I thought about adding it to `lib/datadog/core/logging/ext.rb` at first, but `ddsource` is not tracing specific, it's Datadog Logging specific: it should be added to all log outputs if possible, even without the Tracing product installed.
On the other hand, the remaining fields (e.g. `dd.trace_id, `dd.version`) are tracing<>logs specific.
I can move them to `lib/datadog/core/logging/ext.rb` for now, I see no harm, given it's easy to move them out later.